equinix/resource_metal_device.go

```diff
@@ -625,23 +630,24 @@ func resourceMetalDeviceRead(ctx context.Context, d *schema.ResourceData, meta i
 		return err
 	}
 
-	d.Set("hostname", device.Hostname)
-	d.Set("plan", device.Plan.Slug)
-	d.Set("deployed_facility", device.Facility.Code)
-	d.Set("facilities", []string{device.Facility.Code})
+	d.Set("hostname", device.GetHostname())
+	d.Set("plan", device.Plan.GetSlug())
+	d.Set("deployed_facility", device.Facility.GetCode())
+	d.Set("facilities", []string{device.Facility.GetCode()})
 	if device.Metro != nil {
-		d.Set("metro", device.Metro.Code)
-	}
-	d.Set("operating_system", device.OS.Slug)
-	d.Set("state", device.State)
-	d.Set("billing_cycle", device.BillingCycle)
-	d.Set("locked", device.Locked)
-	d.Set("created", device.Created)
-	d.Set("updated", device.Updated)
-	d.Set("ipxe_script_url", device.IPXEScriptURL)
-	d.Set("always_pxe", device.AlwaysPXE)
-	d.Set("root_password", device.RootPassword)
-	d.Set("project_id", device.Project.ID)
+		d.Set("metro", device.Metro.GetCode())
```

Comment:
Should be safe and equivalent to use `d.Set("metro", device.GetMetro().GetCode())` without the `if` check.